daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/v2/Runner.scala

```diff
@@ -13,187 +13,159 @@ import com.daml.lf.engine.script.ledgerinteraction.{
+      unversionedRunner.extendedCompiledPackages,
+      unversionedRunner.script.expr,
+      iterationsBetweenInterruptions = 100000,
+      traceLog = traceLog,
+      warningLog = warningLog,
+    )(Script.DummyLoggingContext)
 
-    @scala.annotation.tailrec
-    def stepToValue(): Either[RuntimeException, SValue] = {
-      val result = machine.run()
-      canceled() match {
-        case Some(err) => Left(err)
-        case None =>
-          result match {
-            case SResultInterruption =>
-              stepToValue()
-            case SResultFinal(v) =>
-              Right(v)
-            case SResultError(err) =>
-              Left(Runner.InterpretationError(err))
-            case res =>
-              Left(new IllegalStateException(s"Internal error: Unexpected speedy result $res"))
-          }
-      }
+  @scala.annotation.tailrec
+  final def stepToValue(): Either[RuntimeException, SValue] = {
+    val result = machine.run()
+    canceled() match {
+      case Some(err) => Left(err)
+      case None =>
+        result match {
+          case SResultInterruption =>
+            stepToValue()
+          case SResultFinal(v) =>
+            Right(v)
+          case SResultError(err) =>
+            Left(script.Runner.InterpretationError(err))
+          case res =>
+            Left(new IllegalStateException(s"Internal error: Unexpected speedy result $res"))
+        }
     }
+  }
 
-    val initialClientsV1 = initialClients.map(ScriptLedgerClient.realiseScriptLedgerClient)
+  private val initialClientsV1 = initialClients.map(ScriptLedgerClient.realiseScriptLedgerClient)
 
-    val env =
-      new ScriptF.Env(
-        unversionedRunner.script.scriptIds,
-        unversionedRunner.timeMode,
-        initialClientsV1,
-        machine,
-      )
+  private val env =
+    new ScriptF.Env(
+      unversionedRunner.script.scriptIds,
+      unversionedRunner.timeMode,
+      initialClientsV1,
+      machine,
+    )
 
-    def run(
-        expr: SExpr
-    ): Future[SValue] = {
-      machine.setExpressionToEvaluate(expr)
-      stepToValue()
-        .fold(Future.failed, Future.successful)
-        .flatMap(fsu => Converter toFuture unrollFree(fsu))
-        .flatMap {
-          case Right((vv, v)) =>
-            Converter
-              .toFuture(
-                ScriptF.parse(
-                  ScriptF.Ctx(
-                    unversionedRunner.knownPackages,
-                    unversionedRunner.extendedCompiledPackages,
-                  ),
-                  vv,
-                  v,
-                )
+  private val ctx =
+    ScriptF.Ctx(
+      unversionedRunner.knownPackages,
+      unversionedRunner.extendedCompiledPackages,
+    )
+
+  // Wraps the result of executing a command in a call to continue, using the following form
+  // let compute = \() -> resultExpr in let result = compute () in continue result
+  // Note we defer to a lambda in order to maintain ANF
+  private def runCmdQuestion(
```

Comment:
Nice work here to eliminate the continue passing everywhere in ScriptF :)

Reply:
Thank you good sir :) It was quite pleasing